ext/mbstring/mbstring.c

```diff
@@ -2953,6 +2953,57 @@ PHP_FUNCTION(mb_strtolower)
 	RETURN_STR(mbstring_convert_case(PHP_UNICODE_CASE_LOWER, ZSTR_VAL(str), ZSTR_LEN(str), enc));
 }
 
+static zend_string* php_mb_ulcfirst(zend_string *str, php_case_mode mode, const mbfl_encoding *enc)
+{
+	zend_string *first, *second, *head;
+	first = mb_get_substr(str, 0, 1, enc);
+	second = mb_get_substr(str, 1, MBFL_SUBSTR_UNTIL_END, enc);
+	head = mbstring_convert_case(mode, ZSTR_VAL(first), ZSTR_LEN(first), enc);
+	zend_string_release_ex(first, false);
+
+	zend_string *retval = zend_string_concat2(ZSTR_VAL(head), ZSTR_LEN(head), ZSTR_VAL(second), ZSTR_LEN(second));
+	zend_string_release_ex(head, false);
+	zend_string_release_ex(second, false);
+
+	return retval;
+}
+
+PHP_FUNCTION(mb_ucfirst)
```

Comment:
In principle, you could move all of the parameter parsing too to the helper function `php_mb_ulcfirst`, the signature for the arguments of `php_mb_ulcfirst` would then be `php_mb_ulcfirst(INTERNAL_FUNCTION_PARAMETERS, php_case_mode mode)` and to pass the right parameters in the `PHP_FUNCTION`, you could use `INTERNAL_FUNCTION_PARAM_PASSTHRU`

Reply:
True, but I think what @youkidearitai is good.

Reply:
Thank you! That's how I did it.